components/detail/SwapDetailSummary/SwapDetailMobileValue.tsx

```diff
@@ -4,13 +4,16 @@ import dayjs from 'dayjs';
 import utc from 'dayjs/plugin/utc';
 import IconStatus from 'components/common/IconStatus';
 import { StepsIcon } from 'components/icons';
+import Tooltip from 'components/common/Tooltip';
 
 dayjs.extend(utc);
 
 function SwapDetailMobileValue(props: SwapDetailItem) {
   const { details, column } = props;
   const { estimatedTimeInSeconds, status, steps, creationDate } = details;
   const successStep = steps.filter((item) => item.status === 'success').length;
+  const currentStep = details.steps.find((step) => step.status === 'running');
```

Comment:
It always seems to return the last step. Maybe because some steps are unknown.
create new swap on app.rango.exchange that has multi step and check it.
You can also use this link, if it still has an expiration date:
http://localhost:3000/swap/bdbbcd99-dd2f-4124-96af-d769895c6d00

Reply:
fixed